### ISSUE TITLE
Add an interactive diagram content block

### ISSUE THREAD

**Issue by vitorgomateus** [uchicago-library/research_data#49]
## To-do
- [ ] This custom SVG needs to be added as a Wagtail template file.
- [ ] It needs to be included in the home page and standard page via a Boolean field.
- [ ] It should have a settings page to customize with mandatory title and description for accessibility purposes.
- [ ] It should have relevant help text in the Boolean field and in the settings page.

## Bonus To-do
- [ ] Make the 7 links and texts overridable in the settings page, as plain text fields, with help-text on how to format links, and on how text might not fit inside the respective box.
- [ ] Make the fill colors and text colors of the 7 elements overridable in the settings page.
- [ ] Add a field to adjust the font size of `g text{}` between 10 and 18 px (in .5 increments if simple to implement) in the settings page.

## Context
Stakeholders would like an interactive diagram to graphically represent the research lifecycle phase. 
This has concerns on accessibility, responsiveness, maintenance, cost of implementation.

We conducted research on different methods: inline interactive SVG, HTML `<li>` + CSS Positioning, and HTML `<canvas>`.
The conclusion was that a hard-coded SVG could be made accessible and natively interactive without heavy JS manipulation. Using CSS to include hover and active state, using anchor tags inside the SVG to make it natively interactive, and using HTML properties to define `tabindex`, `role`, and `aria-label`.

An SVG file was designed and exported using Adobe Illustrator. Then the code was modified to include all the above considerations. 

### Item order
The items in this diagram should always be considered in this order:

1. Plan and Design phase 
2. Collect and Create phase 
3. Analyze and Collaborate phase 
4. Evaluate and Archive phase 
5. Share phase 
6. Publish and Reuse phase 
7. Store and Manage phase 

## Custom SVG
```html
<section style="display: flex;
        /* flex-direction: column; */
        /* flex-wrap: nowrap; */
        align-items: center;
        justify-content: center;">
        <?xml version="1.0" encoding="UTF-8"?>
        <svg xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:i="http://ns.adobe.com/AdobeIllustrator/10.0/" viewBox="0 0 300 300" role="group" aria-labelledby="lifecycle-title lifecycle-desc" style="
            min-width: 250px;
            max-width: 500px;
            padding: 24px 12px;" >
            <title id="lifecycle-title">Research Data Lifecycle</title>
            <desc id="lifecycle-desc">The research data lifecycle represents the phases of the research data process. The research data lifecycle is one part of the overall research process, and, just as the research process is iterative, so to is the research data lifecycle: each component builds upon the next, and moving between phases is a natural part of the overall research process.</desc>
            <style type="text/css">
                @import url('https://uchicago-brand-fonts.s3.us-east-2.amazonaws.com/UChicagoSansSerif.css');
                g a{
                    text-decoration: none;
                }
                g a:hover path{
                    fill: #000;
                }
                g a:focus path{
                    fill: #000;
                    stroke: #2196F3;
                    stroke-width: 3px;
                }
                g a:hover circle{
                    fill: #a6a6a6;
                }
                g a:focus circle{
                    fill: #a6a6a6;
                    stroke: #2196F3;
                    stroke-width: 3px;
                }
                g text{
                    font-family: "UChicago Sans Serif", "Helvetica Neue", Helvetica, Arial, sans-serif;
                    font-size: 14px;
                    font-weight: 700;
                    letter-spacing: 0.02em;
                    pointer-events: none;
                }
            </style>
            
            <g
                id="plan-and-design-phase" >
                <a href="/research-lifecycle/plan-design/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Plan and Design phase"> 
                <path d="M157.5,40.1l-17.8,44.5c13.8-2.2,28.5,0,41.7,7.1,5.7,3.1,10.7,6.9,15.1,11.2l48.5-7,22.4-39.2c-12.3-15.5-27.8-28.8-46.3-38.8C193.7,3.1,163.6-2.2,134.6.8l22.9,39.3Z" fill="maroon"/>
                <text transform="translate(175.4 53.7)" fill="#fff"><tspan x="0" y="0">Plan &amp;</tspan><tspan x="-2.5" y="16">Design</tspan></text>
                </a>
            </g>
            <g
                id="collect-and-create-phase" >
                <a href="/research-lifecycle/collect-create/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Collect and Create phase"> 
                <path d="M248.9,101.5l-47.4,6.9c13.1,16.2,17.9,37.9,12.5,58.4l30.4,38.5,45.1-.2c19.3-48.5,11.4-102.5-17.9-143.1l-22.6,39.5Z" fill="#a9431e"/>
                <text transform="translate(229.1 144.8)" fill="#fff"><tspan x="0" y="0">Collect</tspan><tspan x="-6.4" y="16">&amp; Create</tspan></text>
                </a>
            </g>
            <g
                id="analyze-and-collaborate-phase" >
                <a href="/research-lifecycle/analyze-collaborate/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Analyze and Collaborate phase"> 
                <path d="M241.4,211.4l-29.7-37.6c-1,2.6-2.2,5.1-3.5,7.7-9,16.6-24,27.8-40.8,32.4l-18.2,45.6,22.7,39c45.1-6.7,86.7-33.9,110-77.1,1.8-3.3,3.4-6.6,4.9-10l-45.5.2Z" fill="#404040"/>
                <text transform="translate(168.6 237.9)" fill="#fff"><tspan x="0" y="0">Analyze &amp;</tspan><tspan x="-7" y="16">Collaborate</tspan></text>
                </a>
            </g>
            <g
                id="evaluate-and-archive-phase" >
                <a href="/research-lifecycle/evaluate-archive/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Evaluate and Archive phase"> 
                <path d="M142.5,259.9l17.8-44.5c-13.8,2.2-28.5,0-41.7-7.1-5.7-3.1-10.7-6.9-15.1-11.2l-48.5,7-22.4,39.2c12.3,15.5,27.8,28.8,46.3,38.8,27.6,14.9,57.7,20.2,86.7,17.2l-22.9-39.3Z" fill="#a9431e"/>
                <text transform="translate(57.3 237.9)" fill="#fff"><tspan x="0" y="0">Evaluate </tspan><tspan x="-1.3" y="16">&amp; Archive</tspan></text>
                </a>
            </g>
            <g
                id="share-phase" >
                <a href="/research-lifecycle/share/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Share phase"> 
                <path d="M51.1,198.5l47.4-6.9c-13.1-16.2-17.9-37.9-12.5-58.4l-30.4-38.5-45.1.2C-8.8,143.3-.9,197.3,28.4,237.9l22.6-39.5Z" fill="#404040"/>
                <text transform="translate(18.7 152.8)" fill="#fff"><tspan x="0" y="0">Share</tspan></text>
                </a>
            </g>
            <g
                id="publish-and-reuse-phase" >
                <a href="/research-lifecycle/publish-reuse/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Publish and Reuse phase"> 
                <path d="M58.6,88.6l29.7,37.6c1-2.6,2.2-5.1,3.5-7.7,9-16.6,24-27.8,40.8-32.4l18.2-45.6L128,1.6C82.9,8.3,41.4,35.5,18,78.7c-1.8,3.3-3.4,6.6-4.9,10l45.5-.2Z" fill="#a9431e"/>
                <text transform="translate(58.8 53.7)" fill="#fff"><tspan x="0" y="0">Publish</tspan><tspan x="-2.2" y="16">&amp; Reuse</tspan></text>
                </a>
            </g>
            <g
                id="store-and-manage-phase" >
                <a href="/research-lifecycle/store-manage/" role="button" 
                role="button" 
                tabindex="0" 
                aria-label="Store and Manage phase"> 
                <circle cx="150" cy="150" r="60.2" fill="#d9d9d9"/>
                <text transform="translate(120.8 144.8)" fill="#000"><tspan><tspan x="0" y="0">Store &amp;</tspan></tspan><tspan><tspan x="-3" y="16">Manage</tspan></tspan></text>
                </a>
            </g>

        </svg>

    </section>
```
